File: part2-refactoring-gym/java/step2-early-return/src/test/java/badcode/DeepUmpireTest.java
package badcode;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import java.util.List;
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class DeepUmpireTest {
    @DisplayName("같은 자리에 같은 수가 있으면 스트라이크이다")
    @Test
    void countStrikeReturnsCorrectCount() {
        // given
        DeepUmpire umpire = new DeepUmpire();
        List<Integer> answer = List.of(1, 2, 3);
        List<Integer> guess = List.of(1, 4, 5);

        // when
        int strike = umpire.countStrike(answer, guess);

        // then
        assertThat(strike).isEqualTo(1);
    }

    @DisplayName("입력값이 null이면 예외가 발생한다")
    @Test
    void throwsExceptionWhenInputIsNull() {
        DeepUmpire umpire = new DeepUmpire();
        assertThatThrownBy(() -> umpire.countStrike(null, List.of(1, 2, 3)))
                .isInstanceOf(IllegalArgumentException.class);
    }

    @DisplayName("입력값의 크기가 다르면 예외가 발생한다")
    @Test
    void throwsExceptionWhenSizeMismatch() {
        DeepUmpire umpire = new DeepUmpire();
        assertThatThrownBy(() -> umpire.countStrike(List.of(1, 2), List.of(1, 2, 3)))
                .isInstanceOf(IllegalArgumentException.class);
    }
}

File: part2-refactoring-gym/ts/step2-early-return/deepUmpire.test.ts
import { DeepUmpire } from './deepUmpire';

describe('DeepUmpire', () => {
    it('정상_동작', () => {
        const umpire = new DeepUmpire();
        const answer = [1, 2, 3];
        const guess = [1, 4, 5];
        const strike = umpire.countStrike(answer, guess);
        expect(strike).toBe(1);
    });

    it('입력값이_null이면_예외발생', () => {
        const umpire = new DeepUmpire();
        expect(() => umpire.countStrike(null, [1, 2, 3])).toThrow('입력값은 null일 수 없습니다.');
    });

    it('입력값의_크기가_다르면_예외발생', () => {
        const umpire = new DeepUmpire();
        expect(() => umpire.countStrike([1, 2], [1, 2, 3])).toThrow('입력값의 길이가 올바르지 않습니다.');
    });
});
